modules/va_facilities/app/controllers/va_facilities/v1/nearby_controller.rb

```diff
@@ -17,31 +16,34 @@ class NearbyController < ApplicationController
       skip_before_action(:authenticate)
       before_action :set_default_format
       before_action :set_facility_type
+      before_action :set_default_drivetime
       before_action :validate_params, only: [:index]
 
+      PER_PAGE = 20
+
       REQUIRED_PARAMS = {
         address: %i[street_address city state zip].freeze,
         lat_lng: %i[lat lng].freeze
       }.freeze
-      QUERY_INFO = {
-        address: NearbyFacility.method(:query),
-        lat_lng: NearbyFacility.method(:query_by_lat_lng)
-      }.freeze
 
       def index
-        query_method = get_query_method(params)
-        params_hash = params.permit!.to_h.symbolize_keys
-        resource = query_method.call(params_hash).paginate(page: params[:page],
-                                                           per_page: params[:per_page] || NearbyFacility.per_page)
+        lat_lng = get_lat_lng(params)
+        eligible_facilities = Facilities::VHAFacility.with_services(params[:services]) if params[:services]
```

Comment:
It wasn't immediately clear to me that `eligible_facilities` is actually a list of ids, and not a facilities relation. I think the select/pluck statement in `with_services` should be moved up into the controller, OR maybe just have `with_services` method renamed to `ids_with_services` to make it extra clear. I also think it should be `eligible_facility_ids`.

Reply:
I like the idea of moving the `pluck` into the controller. That would allow us to theoretically reuse the query itself and return any fields we want.

Reply:
Moved pluck into the controller and made the method on `VHAFacility` return the whole facility.